pyproj/_crs.pyx

```diff
@@ -117,15 +244,165 @@ cdef class AreaOfUse:
+        -------
+        str: The WKT string.
+        """
+        return _to_wkt(self.projctx, self.projobj, version, pretty=pretty)
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.to_wkt(pretty=True)
+
+    def is_exact_same(self, Base other):
+        """Compares projections to see if they are exactly the same."""
+        return proj_is_equivalent_to(
+            self.projobj, other.projobj, PJ_COMP_STRICT) == 1
+
+    def __eq__(self, Base other):
+        """Compares projections to see if they are equivalent."""
+        return proj_is_equivalent_to(
+            self.projobj, other.projobj, PJ_COMP_EQUIVALENT) == 1
+
+
+_COORD_SYSTEM_TYPE_MAP = {
+    PJ_CS_TYPE_UNKNOWN: "unknown",
+    PJ_CS_TYPE_CARTESIAN: "cartesian",
+    PJ_CS_TYPE_ELLIPSOIDAL: "ellipsoidal",
+    PJ_CS_TYPE_VERTICAL: "vertical",
+    PJ_CS_TYPE_SPHERICAL: "spherical",
+    PJ_CS_TYPE_ORDINAL: "ordinal",
+    PJ_CS_TYPE_PARAMETRIC: "parametric",
+    PJ_CS_TYPE_DATETIMETEMPORAL: "datetimetemporal",
+    PJ_CS_TYPE_TEMPORALCOUNT: "temporalcount",
+    PJ_CS_TYPE_TEMPORALMEASURE: "temporalmeasure",
+}
+
+cdef class CoordinateSystem(Base):
+    """
+    Coordinate System for CRS
+
+    Attributes
+    ----------
+    name: str
+        The name of the coordinate system.
+
+    """
+    def __init__(self):
+        self._axis_list = None
+
+    @staticmethod
+    cdef create(PJ_CONTEXT* projcontext, PJ* projobj):
+        cdef CoordinateSystem coord_system = CoordinateSystem()
+        coord_system.projobj = proj_crs_get_coordinate_system(
+            projcontext,
+            projobj
+        )
+        if coord_system.projobj == NULL:
+            return None
+
+        cdef PJ_COORDINATE_SYSTEM_TYPE cs_type = proj_cs_get_type(
+            coord_system.projctx,
+            coord_system.projobj
+        )
+        coord_system.name = _COORD_SYSTEM_TYPE_MAP[cs_type]
+        coord_system.projctx = get_pyproj_context()
+        return coord_system
+
+
+    @property
+    def axis_list(self):
+        """
+        Returns
+        -------
+        list[Axis]: The Axis list for the coordinate system.
+        """
+        if self._axis_list is not None:
+            return self._axis_list
+        self._axis_list = []
+        cdef int num_axes = 0
+        num_axes = proj_cs_get_axis_count(
+            self.projctx,
+            self.projobj
+        )
+        for axis_idx from 0 <= axis_idx < num_axes:
+            self._axis_list.append(
+                Axis.create(
+                    self.projctx,
+                    self.projobj,
+                    axis_idx
+                )
+            )
+        return self._axis_list
+
+
+cdef class Ellipsoid(Base):
+    """
+    Ellipsoid for CRS
+
+    Attributes
+    ----------
+    name: str
+        The name of the ellipsoid.
+    is_semi_minor_computed: int
+        1 if True, 0 if False
+    ellipsoid_loaded: bool
+        True if it is loaded without errors.
```

Comment:
Also add the `semi_major_metre` and `semi_minor_metre` attributes here?

Reply:
Those are documented as properties, so not needed here.